src/detect-tag-pcap.c

```diff
@@ -0,0 +1,291 @@
+
+    if (unlikely(t == NULL)) {
+        FatalError(SC_ERR_TS, "Unable to create time structure, ts-error");
+    }
+
+    char fmt_time_buf[64];
+    CreateFormattedTimeString(t, "%Y%m%d_%H%M%S.%%06u", fmt_time_buf,
+                              sizeof(fmt_time_buf));
+
+    int ret = snprintf(time_buf, time_buf_size,
+                       fmt_time_buf, packet_time);
+    if (ret < 0 || (size_t) ret >= time_buf_size) {
+        SCLogError(SC_ERR_INVALID_NUM_BYTES, "Provided buffer size is too "
+                                             "small to create full time "
+                                             "buffer. Error: %s",
+                   strerror(errno));
+    }
+}
+
+/**
+ *  \brief Fills the result_path_buf with a full file path that can be used
+ *   to create a file. InitializePcapLogFilenameSupport() must be run before
+ *   this is called.
+ *  \param result_path_buf buffer to hold the path string.
+ *  \param result_buf_size length of the filename buffer used.
+ *  \param p packet pointer.
+ *  \param signature that alerted.
+ *  \param thread_id unique id for the current thread.
+ *  \param unique_id counter incrementing over time to add entropy to filenames.
+ */
+void GenerateStreamFilepath(char *result_path_buf, size_t result_buf_size,
+        TaggedPcapEntry *tpe)
+{
+    if (tpe == NULL || result_path_buf == NULL) {
+        return;
+    }
+    char time_buf[FILENAME_TIMEBUF_SIZE];
+    GenerateStreamFilenameTimebuf(time_buf, FILENAME_TIMEBUF_SIZE, tpe->time);
+    int ret;
+
+    ret = snprintf(result_path_buf, result_buf_size,"%s/%u-%u.%s.%u.%s.pcap",
+            g_output_config.g_output_dir, tpe->thread_id, tpe->unique_id,
+            g_output_config.g_hostname, tpe->signature_id, time_buf);
+
+    if (ret < 0 || (size_t)ret >= result_buf_size) {
+        FatalError(SC_ERR_INVALID_NUM_BYTES, "Provided buffer size is too "
+                                             "small to create PCAP file path. "
+                                             "Error: %s", strerror(errno));
+    }
+}
+
+/**
+ *  \brief Registration for TaggedPcapEntry when writing the output to files.
+ *  \param p pointer to the packet being alerted on.
+ *  \param sig signature that triggered the alert.
+ *  \param thread_id id of the thread we're on; used for uniqueness in filename.
+ *  \param unique_id unique counter per thread that increments to increase
+ *  the entropy of filenames.
+ */
+TaggedPcapEntry *SetupTaggedPcap(const Packet *p, const Signature *sig, int
+        thread_id, int unique_id)
+{
+    char pcap_file_path[PATH_MAX];
+    TaggedPcapEntry *tpe = (TaggedPcapEntry*) SCCalloc(1,sizeof(*tpe));
+    if (unlikely(tpe == NULL)) {
+        return NULL;
+    }
+    /*
+     * Initialize tpe's id & time variables
+     */
+    tpe->time = p->ts.tv_usec;
+    tpe->thread_id = thread_id;
+    tpe->unique_id = unique_id;
+    tpe->signature_id = sig->id;
+
+    GenerateStreamFilepath(pcap_file_path, PATH_MAX, tpe);
```

Comment:
this can `FatalError` while we are on the packet path. Can we make it more forgiving?

Reply:
Sure thing! I'll switch it to a SCLogError and then just not open up the pcap_dumper in the case of an error.